src/Forms/UrlField.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+/**
+ * Text input field with validation for a url using filter_var($url, FILTER_VALIDATE_URL)
+ * The value of the field must start with a protocol such as http:// or https://
+ */
+class UrlField extends TextField
+{
+    protected $inputType = 'text';
+
+    public function Type()
+    {
+        return 'text';
+    }
+
+    /**
+     * @param Validator $validator
+     *
+     * @return string
+     */
+    public function validate($validator)
+    {
+        $result = true;
+        $this->value = trim($this->value ?? '');
```

Comment:
Why trim? We should either always trim the value (i.e. trim it in `setValue()`) or never trim it. The value is the value, and it's either valid or not.

Reply:
Removed